pkg/daemon/ceph/client/pool.go

```diff
@@ -409,7 +427,7 @@ func createECPoolForApp(context *clusterd.Context, clusterInfo *ClusterInfo, ecP
 	return nil
 }
 
-func createReplicatedPoolForApp(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec *cephv1.ClusterSpec, pool cephv1.NamedPoolSpec, pgCount, appName string) error {
+func createReplicatedPoolForApp(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec *cephv1.ClusterSpec, pool cephv1.NamedPoolSpec, pgCount, appName, autoscaleMode string) error {
```

Comment:
This signature should be changed. It's not safe to convert values to string when the value should only be limited to what is supported.

```suggestion
func createReplicatedPoolForApp(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec *cephv1.ClusterSpec, pool cephv1.NamedPoolSpec, pgCount, appName, autoscaleMode cephv1.PoolAutoscaleMode) error {
```